src/openassetio-core/include/openassetio/errors.h

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#pragma once
+/**
+ * @file
+ *
+ * Defines core error code constants shared between C and C++.
+ *
+ * To remove any ambiguity in error codes, we want to ensure codes of
+ * different error types (i.e. exceptions vs. batch element errors) do
+ * not overlap.
+ */
+
+/**
+ * First error code used for representing a C++ exception.
+ *
+ * This macro defines the starting error code that should be used for
+ * exception-like errors.
+ */
+#define OPENASSETIO_ErrorCode_BEGIN 1
+
+/**
+ * First error code used for representing a batch element error.
+ *
+ * Assumes a maximum of 100 exception-like error codes before these.
+ */
+#define OPENASSETIO_BatchErrorCode_BEGIN (OPENASSETIO_ErrorCode_BEGIN + 100)
```

Comment:
[question] Is there any merit in making this some power of 2?

Reply:
:thinking: Interesting. Off the top of my head, we _could_ use bit-masking/shifting instead of inequality to test if a given error code is exception-like or batch element-like, but the additional boilerplate makes that not so useful. 

Anything else you can think of?

Reply:
Yeah I think that is what was lurking around in my mind - that it would might make it easier to work out what 'class' of code it was? But as they're never meant to be used in the same context - do we think we'd ever have a use case?

Reply:
Doesn't do any harm, tweaked in 84f7f84a